dbt-bigquery/src/dbt/adapters/bigquery/credentials.py

```diff
@@ -219,12 +235,69 @@ def _create_google_credentials(credentials: BigQueryCredentials) -> GoogleCreden
             scopes=credentials.scopes,
         )
 
+    elif credentials.method == _BigQueryConnectionMethod.EXTERNAL_OAUTH_WIF:
+        creds = _create_identity_pool_credentials(credentials=credentials)
+
     else:
         raise FailedToConnectError(f"Invalid `method` in profile: '{credentials.method}'")
 
     return creds
 
 
+def _create_identity_pool_credentials(credentials: BigQueryCredentials) -> GoogleCredentials:
+    """
+    The ADC dict here represents a configuration for a Google Cloud credential that is used to authenticate with a Google Cloud service.
```

Comment:
Is ADC a standard acronym? I think I'd appreciate a "ADC (<whatever adc standards for>)" type comment here